### PR TITLE
Update SWIG version, cleaner setup and fix PoCL installation

### PR DESCRIPTION
**Pull request by StrikerRUS**
Cross-reference: https://github.com/microsoft/LightGBM/pull/6541.

### CODE REVIEW

File: images/manylinux_2_28_x86_64/Dockerfile [hunk 1]
```diff
@@ -91,31 +92,34 @@ RUN git clone --depth 1 --branch v1.8 https://github.com/pocl/pocl.git \
     -DENABLE_POCLCC=OFF \
     -DENABLE_TESTS=OFF \
     -DENABLE_EXAMPLES=OFF \
+    -DLLC_HOST_CPU=generic \
```
Comment (StrikerRUS) [inline images/manylinux_2_28_x86_64/Dockerfile]: Fix the following error:
```
#9 10.85 -- Find out LLC target triple (for host x86_64-unknown-linux-gnu)
#9 10.86 -- Find out LLC host CPU with /usr/local/bin/llc
#9 10.87 CMake Error at cmake/LLVM.cmake:721 (message):
#9 10.87   LLVM could not recognize your CPU model automatically.  Please run CMake
#9 10.87   with -DLLC_HOST_CPU=<cpu> (you can find valid names with: llc -mcpu=help)
#9 10.87 Call Stack (most recent call first):
#9 10.87   CMakeLists.txt:472 (include)
#9 10.87 
#9 10.87 
#9 10.87 -- Configuring incomplete, errors occurred!
#9 10.87 See also "/pocl/build/CMakeFiles/CMakeOutput.log".
```

**Comment by StrikerRUS** [comment]
> tests/c_api_test/test_.py ./__w/1/s/.ci/test.sh: line 288:  1422 Aborted                 (core dumped) pytest ./tests
https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=16615&view=logs&j=9ce42e5d-f31a-544a-a6de-dc42f8c013ed&t=b696def5-78ae-590c-4234-04332342ba2d&l=470

**Comment by StrikerRUS** [comment]
Works fine!

> -- Found SWIG: /usr/local/bin/swig4.0 (found version "4.2.1") 
https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=16620&view=logs&j=535e9539-0564-5d34-2431-da2644b83c50&t=7cfc64df-bff4-5eef-5deb-8ea58a924da8&l=52

File: images/manylinux_2_28_x86_64/Dockerfile (hunk 1 above)
Comment (StrikerRUS) [reply images/manylinux_2_28_x86_64/Dockerfile]: `generic` results in [`Aborted (core dumped)`](https://github.com/guolinke/lightgbm-ci-docker/pull/34#issuecomment-2227394537). `x86-64` seems to be OK.

Comment (jameslamb) [reply images/manylinux_2_28_x86_64/Dockerfile]: Makes sense to me based on the description at https://portablecl.org/docs/html/install.html

> _`-DLLC_HOST_CPU=<something>` Defaults to auto-detection via `llc`. Run `llc -mcpu=help` for valid values. The CPU type is required to compile the “target” (kernel library) part of CPU backend._

> _This variable overrides LLVM’s autodetected host CPU at configure time. Useful when llc fails to detect the CPU (often happens on non-x86 platforms, **or x86 with CPU newer than LLVM**)._

For as long as we are still using pocl 1.8 (https://github.com/microsoft/LightGBM/issues/5596), LLVM is pinned to v11:

https://github.com/guolinke/lightgbm-ci-docker/blob/a0df140729a4006b4ff87b880e7aa53934ecc0e7/images/manylinux_2_28_x86_64/Dockerfile#L17-L23

LLVM 11.1.0 came out in February 2021 ([release history](https://releases.llvm.org/)), so I guess it's possible that GitHub's hosted runners could have a newer CPU than that.

Comment (jameslamb) [reply images/manylinux_2_28_x86_64/Dockerfile]: Thank you for the investigation!

**Review by jameslamb — APPROVED** [review]
great, thank you!

**Pull request merged**